subsys/net/lib/nrf_cloud/src/nrf_cloud_codec.c

```diff
@@ -889,6 +891,252 @@ static int nrf_cloud_encode_service_info_ui(const struct nrf_cloud_svc_info_ui *
 	return 0;
 }
 
+#ifdef CONFIG_MODEM_INFO
+
+#define MODEM_INFO_JSON_KEY_NET_INF	"networkInfo"
+#define MODEM_INFO_JSON_KEY_SIM_INF	"simInfo"
+#define MODEM_INFO_JSON_KEY_DEV_INF	"deviceInfo"
+
+static int modem_info_add_data(struct lte_param *param, cJSON *json_obj)
+{
+	char data_name[MODEM_INFO_MAX_RESPONSE_SIZE];
+	enum at_param_type data_type;
+	int ret;
+
+	__ASSERT_NO_MSG(param != NULL);
+	__ASSERT_NO_MSG(json_obj != NULL);
+
+	memset(data_name, 0, ARRAY_SIZE(data_name));
+	ret = modem_info_name_get(param->type,
+				data_name);
+	if (ret < 0) {
+		LOG_DBG("Data name not obtained: %d", ret);
+		return -EINVAL;
+	}
+
+	data_type = modem_info_type_get(param->type);
+	if (data_type < 0) {
+		return -EINVAL;
+	}
+
+	if (data_type == AT_PARAM_TYPE_STRING &&
+	    param->type != MODEM_INFO_AREA_CODE) {
+		ret = json_add_str_cs(json_obj, data_name, param->value_string);
+		if (ret) {
+			return ret;
+		}
+	} else {
+		ret = json_add_num_cs(json_obj, data_name, param->value);
+		if (ret) {
+			return ret;
+		}
+	}
+
+	return ret;
+}
+
+static int modem_info_network_encode(struct network_param *network, cJSON *json_obj)
```

Comment:
The function name should have a different prefix than those from modem_info to avoid any confusion.

Reply:
✔